test_run.lua

```diff
@@ -284,6 +285,26 @@ local function grep_log(self, node, what, bytes, opts)
     return found
 end
 
+-- Checks fn with delay until timeout or fn returns true, otherwise returns false
+local function wait_cond(self, fn, timeout, delay)
+    assert(timeout ~= nil)
+    delay = delay or 0.001
+    local t1 = clock.monotonic()
+    while not fn() do
+        local work_time = clock.monotonic() - t1
+        if work_time > timeout then
+            return false
+        end
+        fiber.sleep(delay)
+    end
+    return true
```

Comment:
I think it would be good to return the value the function `fn` returns. It is worth to do as in the success case as well as in the fail case (it can be nil or false).